R/plots.R

```diff
@@ -0,0 +1,241 @@
+line_chart <- function(...) UseMethod ("line_chart")
+
+#' @param df dataframe. The dataframe use
+#' @param x string. The name of the x variable.
+#' @param y string. The name of the y variable.
+#' @param color string. The name of the color variable.
+#' @param title string. The title for the plot
+#' @param x_lab string. The label for the x-axis.
+#' @param y_lab string. The label for the y-axis.
+#' @param legend_lab string. The label for the legend.
+#' @param annotation_text_size integer. The size for the text.
+#' @param label string. The name of the label.
+line_chart.default <- function(df = NULL,
+                               x = NULL,
+                               y = NULL,
+                               color = NULL,
+                               title = NULL,
+                               x_lab = NULL,
+                               y_lab = NULL,
+                               legend_lab = NULL,
+                               annotation_text_size = NULL,
+                               label = NULL,
+                               usafacts_source = "Confirmed COVID-19 cases and deaths: USAFacts Data (https://usafacts.org/)",
+                               census_source = "Population data: 2018 American Community Survey 5-year Estimates",
+                               covid_tracking_source = "Testing data: The Covid Tracking Project (https://covidtracking.com/)",
+                          ...) {
+# browser()
+  x_vec <- df[[x]]
+  y_vec <- df[[y]]
+
+  if(!is.null(color)) color_vec <- df[[color]]
+  if(!is.null(label)) label_vec <- df[[label]]
+
+  if(is.null(color)) {
+
+    if(is.null(label)) {
+      gg <- ggplot(df, aes(x = x_vec, y = y_vec))
+        geom_point(color = "#e54e4d")
+
+    } else{
+      
+      gg <- ggplot(df, aes(x = x, y = y, label = str_wrap(label_vec, width = 65))) +
+        geom_point(color = "#e54e4d") +
+        geom_text_repel(size = 3,
+                        hjust = 0,
+                        nudge_x = 2)
```

Comment:
I also, on my end, changed these to geom_labels. I'm not sure if its better or worse but may be something to play around with

Reply:
Yes, let's definitely try to add a toggle. I think that's an excellent idea.

Reply:
The other thing I want to do was add different timeline markings in the background, then have the timelines appear as a legend rather than "WHO declares COVID-19 pandemic" as text on the graph. I think that will help them look a little cleaner as well.

Reply:
Ohhh I like this. Basically I think we would create a gridtree/series of grobs, with one being the main plot and one being the timeline (below or above it).

How it would work: https://cran.r-project.org/web/packages/gridExtra/vignettes/arrangeGrob.html
Examples similar: https://cran.r-project.org/web/packages/ggplotify/vignettes/ggplotify.html

We could also look into using Patchwork https://github.com/thomasp85/patchwork

Reply:
(Just brainstorming) what could be cool would be two interactive plots, one being the actual plot and one being a pseudo-timeline of key events. Hovering over the points in the top outputs deaths, confirmed, tests, etc. Hovering over the bottom 'key events' expands them into a larger tooltip (for longer text).